### DISCUSSION TITLE
How to reapply traefik install in existing cluster?

### DISCUSSION THREAD

**Discussion by jeff1985** [mysticaltech/terraform-hcloud-kube-hetzner#798] [Q&A] [answered]
I have accidentially deleted the traefik deployment. How can I recreate it?

I tried changing _traefik_values_, but the deployment object still is not created after terraform apply. Any ideas?

**Comment by mysticaltech** [comment, marked as answer]
@jeff1985 You need to delete the helm chart and apply again. Grab it with `kubectl get helmchart -A` to find the exact name, and then dump it with -o yaml, then delete and apply again (all with kubectl).

**Reply by jeff1985** [reply]
Thanks for clarification! 

I ended with extracting the values from helmchart to separate file and then installed the helm release via terraform:
   resource "helm_release" "traefik" {
      name = "traefik"
      chart = "traefik"
    
      repository = "https://traefik.github.io/charts"
      namespace = "traefik"
    
      values = [
        file("${path.module}/traefik-values.yaml")
      ]
    }

This gives me the flexibility to edit any value and then have terraform reapply the values or recreate the release if needed.

**Reply by mysticaltech** [reply]
Great, also keep in mind that you can use helmchatconfig definitions too. https://docs.k3s.io/helm
